### PR TITLE
Added a fuzzer for SMIME

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by kroeckx** [comment]
Can you also make it run during make test, see for instance #20269 on how to do it.

File: fuzz/smime.c
Comment (kroeckx) [inline fuzz/smime.c:28]: Can you do more things with the PKCS7? For instance call functions like PKCS7_get_signer_info(), PKCS7_cert_from_signer_info(), PKCS7_get_signed_attribute(), PKCS7_get_smimecap(), PKCS7_dataVerify(), ...

Comment (kroeckx) [reply fuzz/smime.c:28]: I guess the CMS fuzzer could use things like that too.

Comment (alex) [reply fuzz/smime.c:32]: Added some usage of the p7 API. I'm candidly not totally confident they're correct, since many of these functions aren't documented.

Comment (kroeckx) [reply fuzz/smime.c:34]: You might want to look at the pkcs7 and smime tool for other things that might be useful to do.

**Comment by alex** [comment]
@kroeckx do you want to hold this open until its expanded, or would you consider merging as is (modulo any other reviews) and then iterating as we get coverage metrics from oss-fuzz?

**Comment by kroeckx** [comment]
I'm fine with merging as is.

File: fuzz/smime.c
Comment (kroeckx) [inline fuzz/smime.c:24]: We have a 4 space style

**Review by kroeckx — APPROVED** [review]

File: fuzz/smime.c
Comment (alex) [reply fuzz/smime.c:24]: fixed

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:17]: Nit: For functions only we always place the opening `{` on the next line. Here and in other functions below. Also the "return 1;" should be a line on its own (not on same line as `{}`)

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:44]:
```` C
void FuzzerCleanup(void)
{
}
````

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:23]: `if (p7 != NULL) {`

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:25]: Please insert a blank line after declarations

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:27]: No space after `*`

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:32]: The `if` isn't necessary here. Just call `sk_X509_ALGOR_pop_free` anyway. It tolerates the NULL.

File: test/recipes/99-test_fuzz_smime.t
Comment (mattcaswell) [inline test/recipes/99-test_fuzz_smime.t:2]: 2023

File: fuzz/smime.c
Comment (mattcaswell) [inline fuzz/smime.c:2]: 2023

**Review by mattcaswell — CHANGES_REQUESTED** [review]
Looks good. Just some style nits below.

**Review by mattcaswell — APPROVED** [review]
LGTM. @kroeckx - can you reconfirm?

**Review by kroeckx — APPROVED** [review]

**Comment by alex** [comment]
is anything additionally required from me to merge this?

**Comment by kroeckx** [comment]
No, it will get merged when someone has time for it.

**Comment by mattcaswell** [comment]
Pushed. Thanks for your contribution.

**Pull request closed without merging**
